src/iuse_actor.cpp

```diff
@@ -2583,6 +2599,16 @@ repair_item_actor::attempt_hint repair_item_actor::repair( player &pl, item &too
     if( action == RT_REPAIR ) {
         if( roll == SUCCESS ) {
             pl.add_msg_if_player(m_good, _("You repair your %s!"), fix.tname().c_str());
+            if( t->has_quality( quality_sew ) ) {
+                p->add_msg_if_player(m_info,
+                    _("You have used %d filament charges."), filament_required);
+                for (const auto &it : req.get_components()) {
+                    p->consume_items(it);
+                }
+                for (const auto &it : req.get_tools()) {
+                    p->consume_tools(it);
+                }
```

Comment:
This one is harder than that: you'd need to somehow remember which component player used last time.
It is not acceptable for the game to ask player which filament to use on every "tick" of repair.

This is rather complicated. If you're sure you want to do it, better try it in a separate PR.